File: utils/api_call.py
import requests

class API_call():
    def __init__(self, base_url, resource, headers={"Content-Type": "application/json"}):
        self.base_url = base_url + "/" + resource
        self.headers = headers

    def call(self, type, timeout, *params):
        url = self.base_url + "?"
        for param in params:
            url += param[0] + "=" + param[1] + "&"
        url = url[:-1]
        print("Request url: " + url)
        if type.upper() == "GET":
            response = requests.get(url, headers=self.headers, timeout=timeout)
        if type.upper() == "PUT":
            response = requests.put(url, headers=self.headers, timeout=timeout)
        if type.upper() == "POST":
            response = requests.post(url, headers=self.headers, timeout=timeout)
        if type.upper() == "DELETE":
            response = requests.delete(url, headers=self.headers, timeout=timeout)
        print("Status code: " + str(response.status_code))
        if response.status_code == 200:
            json_response = response.json()
            print("Message: " + json_response["message"])
            return json_response
        return {"message": "Request failed", "error": True}

File: HRI/tablet/javascript_utils/api_call.js
class ApiClient {
    constructor(baseURL) {
        this.baseURL = baseURL;
    }

    async call(method, endpoint, data = null, params = null) {
        const url = `${this.baseURL}${endpoint}${params ? `?${new URLSearchParams(params).toString()}` : ''}`;
        console.log(url)
        const options = {
            method,
            headers: {
                'Content-Type': 'application/json',
                "Access-Control-Allow-Origin": this.baseURL
            },
        };

        if (data) {
            options.body = JSON.stringify(data);
        }

        try {
            const response = await fetch(url, options);
            const responseData = await response.json();

            if (!response.ok) {
                throw new Error(responseData.message || 'Errore nella chiamata API');
            }

            return responseData;
        } catch (error) {
            throw error;
        }
    }

    async get(endpoint, params) {
        return await this.call('GET', endpoint, null, params);
    }

    async post(endpoint, data, params) {
        return await this.call('POST', endpoint, data, params);
    }

    async put(endpoint, data, params) {
        return await this.call('PUT', endpoint, data, params);
    }

    async delete(endpoint, params) {
        return await this.call('DELETE', endpoint, null, params);
    }
}